src/components/PercentageText.tsx

```diff
@@ -40,7 +39,7 @@ export const PercentageText: VFC<PercentageTextProps> = ({
         alignItems="center"
         columnGap="3px"
       >
-        {percentageData || "--"}%
+        {isDataZero || percentageData ? percentageData : "--"}%
```

Comment:
I'd put the `isDataZero || percentageData` into their own boolean const to improve readability of this line

```ts
const valueExists:boolean = isDataZero || percentageData

...
{valueExists ? percentageData : "--"}%
...
```